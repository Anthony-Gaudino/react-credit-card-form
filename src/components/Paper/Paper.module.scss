@use "../../theme.scss";

.paper {
  background-color: theme.$white;
  border-radius: 8px;
  box-shadow:
    0 calc(5px * var(--elevation)) calc(10px * var(--elevation)) 0
    theme.$shadow-color;
}
