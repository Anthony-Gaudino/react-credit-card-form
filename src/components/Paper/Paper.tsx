import styles from "./Paper.module.scss";

interface PaperProps {
  /**
   * Paper elevation.
   * Higher elevations has larger shadows.
   */
  elevation?: number;
  /**
   * Paper additional classes.
   */
  className?: string;
  /**
   * Paper children.
   */
  children?: React.ReactNode;
}

/**
 * A Google Material style paper container.
 */
export const Paper = (props: PaperProps): JSX.Element => {
  const { elevation = 0, className: classes, children } = props;

  const style = { "--elevation": elevation } as React.CSSProperties;

  return (
    <div className={`${styles.paper} ${classes}`} style={style}>
      {children}
    </div>
  );
};
