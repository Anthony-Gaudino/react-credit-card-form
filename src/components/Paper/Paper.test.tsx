import { render, screen } from "@testing-library/react";

import Paper from ".";

describe("Test Paper component", () => {
  test("the container is correctly rendered", () => {
    const { asFragment } = render(
      <Paper>
        <span>Paper content</span>
      </Paper>
    );

    const paperContent = screen.getByText("Paper content");

    expect(paperContent).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
