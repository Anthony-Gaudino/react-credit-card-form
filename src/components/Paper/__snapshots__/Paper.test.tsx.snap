// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Paper component the container is correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="paper undefined"
    style="--elevation: 0;"
  >
    <span>
      Paper content
    </span>
  </div>
</DocumentFragment>
`;
